bundles/org.openhab.core.io.rest/src/main/java/org/eclipse/smarthome/io/rest/internal/resources/RootResource.java

```diff
@@ -90,45 +116,70 @@ private RootBean getRootBean() {
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     public void addRESTResource(RESTResource resource) {
         restResources.add(resource);
+
+        scheduleReload();
     }
 
     public void removeRESTResource(RESTResource resource) {
         restResources.remove(resource);
+        scheduleReload();
     }
 
-    @SuppressWarnings({ "rawtypes", "unchecked" })
     @Activate
-    public void activate() {
-        Configuration configuration;
-        try {
-            configuration = configurationAdmin.getConfiguration(Constants.JAXRS_CONNECTOR_CONFIG, null);
-
-            if (configuration != null) {
-                Dictionary properties = configuration.getProperties();
+    public void activate(Map<String, Object> properties) {
+        restRoot = (String) properties.getOrDefault(Constants.ROOT_PROPERTY, RESTConstants.REST_URI);
 
-                if (properties == null) {
-                    properties = new Properties();
-                }
+        servletContainer = new ServletContainer(createConfig());
 
-                String rootAlias = (String) properties.get(Constants.JAXRS_CONNECTOR_ROOT_PROPERTY);
-                if (!RESTConstants.REST_URI.equals(rootAlias)) {
-                    properties.put(Constants.JAXRS_CONNECTOR_ROOT_PROPERTY, RESTConstants.REST_URI);
+        try {
+            Hashtable<String, String> initParams = new Hashtable<>();
+            initParams.put("com.sun.jersey.api.json.POJOMappingFeature", "false");
+            initParams.put(ServletProperties.PROVIDER_WEB_APP, "false");
+            httpService.registerServlet(restRoot, servletContainer, initParams, null);
+            logger.info("openHAB REST started on {}", restRoot);
+        } catch (ServletException | NamespaceException e) {
+            logger.warn("Could not start REST API on {}: {}", restRoot, e.getMessage(), e);
```

Comment:
That depends if we see REST as essential. But yeah I'll change that

Reply:
My understanding has been that the root resource is exactly used for this purpose (REST).
Does it provide any other stuff, so it would make sense to mark it `active` if the `ServletException` or `NamespaceException` has been thrown. So, does the service make sense without the servlet registration succeeded?